// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package publisher;

// The greeting service definition.
service PubSubService {
    rpc SubscribeOne(SubscribeOneRequest) returns (SubscribeOneReply) {}
    rpc SubscribeTwo(SubscribeTwoRequest) returns (SubscribeTwoReply) {}
    rpc SubscribeThree(SubscribeThreeRequest) returns (SubscribeThreeReply) {}
    rpc SubscribeFour(SubscribeFourRequest) returns (stream SubscribeFourReply) {}
}

message SubscribeOneRequest {}
message SubscribeTwoRequest {}
message SubscribeThreeRequest {}
message SubscribeFourRequest {}

message SubscribeOneReply {}
message SubscribeTwoReply {}
message SubscribeThreeReply {}
message SubscribeFourReply {}
